docs/building_pipelines.rst

```diff
@@ -293,3 +294,423 @@ Warnings raised will also be logged to the error log but will not cause data to
+
+Command Line Interface
+~~~~~~~~~~~~~~~~~~~~~~
+
+A command line interface is provided to access the pipeline debugger. The interface can be started by running the *fledge* script that is used to start, stop and monitor Fledge or by calling the *scripts/debug/debug* script directly.
+
+.. note::
+
+   Due to the sensitive nature of pipeline debugging only user with edit permissions may use this interface. Additionally authentication must be enabled on the Fledge instance for this interface to work.
+
+Starting the Debugger
+#####################
+
+The *debug* command should be given the name of a south or north service to debug.
+
+.. code-block:: bash
+
+   $ scripts/fledge -u admin debug Lathe
+   Debug: Lathe$ 
+
+Upon starting a new prompt will be displayed, if the service is a valid service to be debugged and is running. The user may now enter the various debug commands at this prompt. To complete the debugging session type *exit* or Control-D.
+
+To see a list of available commands enter the command *commands*
+
+.. code-block:: bash
+
+   Debug: Lathe$ commands
+   attach:              Attach the pipeline debugger
+   buffer:              Return the contents of the buffers at every pipeline element
+   detach:              Detach the debugger from the pipeline
+   isolate:             Isolate the pipeline from the destination
+   replay:              Replay the buffered data through the pipeline
+   resumeIngest:        Resume the flow of data into the pipeline
+   setBuffer:           Set the number of readings to hold in each buffer, passing an integer argument
+   state:               Return the state of the debugger
+   step:                Allow readings to flow into the pipeline. Passed an optional number of readings to ingest; default to 1 if omitted
+   store:               Allow data to flow out of the pipeline into storage
+   suspendIngest:       Suspend the ingestion of data into the pipeline
+   Debug: Lathe$ 
+
+Attach
+######
+
+The first command to issue is usually the *attach* command. It attaches the debugger session to the service and begins the process of collecting data at each node within the pipeline. Until the debugger is attached it is not possible to view the data in the pipeline.
+
+.. code-block:: bash
+
+   Debug: Lathe$ attach
+   {
+     "status": "ok"
+   }
+   Debug: Lathe$
+
+The data returned from the *attach* command is the status of running that command encoded within a JSON document.
+
+At this point the debugger is attached and collecting data at each node within the pipeline. By default only one reading is retained at each point in the pipeline, this s the last reading seen at that point in the pipeline.
```

Comment:
- each point in the pipeline, -> each point in the pipeline;
- this s the last reading seen -> this is the last reading seen